### PR TITLE
fix(env-check): remove credentials from zip file

### PR DESCRIPTION
**Pull request by cianmSAP**
#792 

Remove unencrypted credentials from zip file
Fix escaping of system name in markdown

### CODE REVIEW

**Review by devinea — DISMISSED** [review]
Changes and tests look good.

File: packages/environment-check/src/output/zip.ts [hunk 1]
```diff
@@ -20,6 +20,27 @@ function byteNumberToSizeString(byteNumber: number): string {
     return `${parseFloat((byteNumber / Math.pow(1024, i)).toFixed(2))} ${units[i]}`;
 }
 
+/**
+ * Convert results to JSON and removes credentials from (stored system) endpoints.
+ *
+ * @param results - environment check results
+ * @returns json string
+ */
+function convertResultsToJson(results: EnvironmentCheckResult): string {
```
Comment (Klaus-Keller) [inline packages/environment-check/src/output/zip.ts:29]: @cianmSAP, any chance we remove the credential at the time we collect the data so they are never part of the results?

Comment (cianmSAP) [reply packages/environment-check/src/output/zip.ts]: Made updates to never return credentials from the exposed functions

File: packages/environment-check/src/checks/stored-system.ts [hunk 2]
```diff
@@ -23,6 +24,21 @@ export async function checkStoredSystem(storedSystem: Endpoint): Promise<{
 }> {
     const logger = getLogger();
 
+    // retrieve system credentials
+    const storeService = await getService({ logger, entityName: 'system' });
```
Comment (Klaus-Keller) [inline packages/environment-check/src/checks/stored-system.ts:28]:
```suggestion
    const storeService = await getService<BackendSystem, BackendSystemKey>({ logger, entityName: 'system' });
```

File: packages/environment-check/src/checks/stored-system.ts [hunk 3]
```diff
@@ -23,6 +24,21 @@ export async function checkStoredSystem(storedSystem: Endpoint): Promise<{
...
+    const backendKey = new BackendSystemKey({ url: storedSystem.Url, client: storedSystem.Client });
+    const sapSystem = (await storeService.read(backendKey)) as BackendSystem;
```
Comment (Klaus-Keller) [inline packages/environment-check/src/checks/stored-system.ts:30]:
```suggestion
    const sapSystem = await storeService.read(backendKey);
```

**Review by Klaus-Keller — CHANGES_REQUESTED** [review]
This
```
const url = new URL(s.url).origin;
```
https://github.com/SAP/open-ux-tools/pull/829/files#diff-6ebfe3548141e9d4ea15436c74a18fe19e941f3fe4f0c08336a548f6c61e9023R122
leads to problems when you have a stored system with url that contains get parameters.

File: packages/environment-check/src/checks/stored-system.ts (hunk 2 above)
Comment (cianmSAP) [reply packages/environment-check/src/checks/stored-system.ts]: Updated

File: packages/environment-check/src/checks/stored-system.ts (hunk 3 above)
Comment (cianmSAP) [reply packages/environment-check/src/checks/stored-system.ts]: Updated

**Review by Klaus-Keller — APPROVED** [review]
- code looks good
- test coverage is ok
- I've built and tested locally using CLI and no user passwords are in the results


Approved from my side, thanks @cianmSAP!

**Review by devinea — APPROVED** [review]
Thanks @cianmSAP

**Pull request merged**